src/mgr/Mgr.cc

```diff
@@ -756,6 +763,23 @@ bool Mgr::got_mgr_map(const MgrMap& m)
   return false;
 }
 
+bool Mgr::exceeded_initialization_retires() const
+{
+  uint64_t max_retries = g_conf().get_val<uint64_t>("mgr_module_load_max_retries");
+  return initialization_retries > max_retries;
+}
+
+void Mgr::update_initialization_retries()
+{
+  std::lock_guard l(lock);
+  if (!is_initialized()) {
+    initialization_retries++;
```

Comment:
I don't really like the connection between the beacon period and this "retry" counter. The code is not really retrying to load the modules either. I would prefer a simple `ceph::coarse_mono_time Mgr::initialization_started`. Then force active when we exceed some threshold `mgr_module_load_expiration` (perhaps).